### PR TITLE
Fix lint errors from golangci-lint

### PR DESCRIPTION
**Pull request by jsoverson**
This PR:
- Fixes lint errors associated with running `golangci-lint` version `1.50.1`
- Adds `golangci-lint run` to `just test`
- Adds a step to setup `golangci-lint` in CI

This PR mostly adds logging for unchecked errors and removes/comments out unused code as WIP w/TODOs. It's not very aggressive with the default configuration and seemed like a quick win.

### CODE REVIEW

File: pkg/transport/http/router/oauth2/oauth2.go [hunk 1]
```diff
@@ -185,7 +187,9 @@ func (o *Auth) callback(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	setSessionCookie(w, token, claims)
+	if err := setSessionCookie(w, token, claims); err != nil {
+		logger.Error("Could not set session cookie", zap.Error(err))
```
Comment (pkedy) [inline pkg/transport/http/router/oauth2/oauth2.go:191]: If we have a `logger` package it should not leak the underlying implementation (in this case zap). I propose, like in logr, to have `logger.Error` take `err error` as the first parameter which nil being valid/accepted.

File: pkg/transport/httprpc/httprpc.go [hunk 2]
```diff
@@ -230,5 +233,7 @@ func (t *HTTPRPC) handleError(err error, codec channel.Codec, req *http.Request,
 		fmt.Fprint(w, "unknown error")
 	}
 
-	w.Write(payload)
+	if _, error :=w.Write(payload); error != nil {
```
Comment (pkedy) [inline pkg/transport/httprpc/httprpc.go:236]: nit: run `go fmt` on this file.

**Review by pkedy — COMMENTED** [review]
A few comments. Mainly, the `zap` package should not leak and require imports throughout our code.

File: pkg/transport/httprpc/httprpc.go (hunk 2 above)
Comment (jsoverson) [reply pkg/transport/httprpc/httprpc.go]: Fixed and I added a test to fail PRs if files aren't formatted going forward.

File: pkg/transport/http/router/oauth2/oauth2.go (hunk 1 above)
Comment (jsoverson) [reply pkg/transport/http/router/oauth2/oauth2.go]: Removed them all.

**Review by pkedy — APPROVED** [review]

**Pull request merged**